Q and a barebones

ftperez11 opened the pull request.

ftperez11 committed:
created question modal skeleton

ftperez11 committed:
upload mediacard skeleton

ftperez11 committed:
got image upload and rendering working

ftperez11 committed:
question skeleton

ftperez11 committed:
question and answer skeletons working and pics uploading in answer modal

ftperez11 committed:
q and q list rendering with pictures

ftperez11 committed:
load more answers and questions working

ftperez11 committed:
search bar dynamically rendering data

ftperez11 committed:
more added questions renders two at a time, answers do not

ftperez11 committed:
first draft commit

DavidDavydovTech commented on package.json:
```diff
@@ -17,13 +17,15 @@
     "react-router": "^5.2.0",
     "redux": "^4.0.5",
-    "redux-thunk": "^2.3.0"
+    "redux-thunk": "^2.3.0",
```
Needs to be removed later

DavidDavydovTech commented on src/App.js:
```diff
@@ -1,19 +1,19 @@
   render() {
     const { name } = this.props;
     return (
       <>
```
You shouldn't have needed to do this- master was updated near the start of the project to have all imports.

Commit:
Removed App.js

Commit:
Merge branch 'master' of https://github.com/HRNYC32-Ubran-Journey/project-cat-walk into q-and-a-barebones

DavidDavydovTech commented on src/components/qa/subcomponents/UploadMediaCard.jsx:
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import Card from '@material-ui/core/Card';
+import CardMedia from '@material-ui/core/CardMedia';
+import clothingImg from '../../../../dist/clothing.jpg';
```
Unused import. It also doesn't exist in your pull request, so it causes a fatal error.

DavidDavydovTech commented on src/components/qa/subcomponents/UploadMediaCard.jsx:
```diff
@@ -0,0 +1,30 @@
+import CardMedia from '@material-ui/core/CardMedia';
+import clothingImg from '../../../../dist/clothing.jpg';
+
+export default function UploadMediaCard({files}) {
```
Files is missing in props validation

DavidDavydovTech commented on src/components/qa/subcomponents/UploadMediaCard.jsx:
```diff
@@ -0,0 +1,30 @@
+          }}
+        />
+      </Card>
+    )
+  })
```
Missing semicolins

DavidDavydovTech commented on src/components/qa/subcomponents/UploadMediaCard.jsx:
```diff
@@ -0,0 +1,30 @@
+    )
+  })
+  return (
+    <div id="media-cards" style={{ display: 'inline-flex'}}>
```
Missing a space on the right of `display: 'inline-flex'`

DavidDavydovTech commented on src/components/qa/subcomponents/UploadPics.jsx:
```diff
@@ -0,0 +1,63 @@
+    // console.log(files[0])
+  };
+
+  const renderNewUploadPhoto = selectedFile.length <= 5 ? <div><input
+          accept="image/*"
+          id="icon-button-file"
+          type="file"
+          style={{ display: 'none' }}
+          onChange={fileSelectedHandler}
+        />
+        <label htmlFor="icon-button-file">
+          <IconButton
+            color="primary"
+            aria-label="upload picture"
+            component="span"
+          >
+            <PhotoCamera />
+          </IconButton>
+        </label></div> : null;
```
Unused variable.

DavidDavydovTech commented on src/components/qa/subcomponents/UploadPics.jsx:
```diff
@@ -0,0 +1,63 @@
+import PhotoCamera from '@material-ui/icons/PhotoCamera';
+import UploadMediaCard from './UploadMediaCard';
+
+export default function UploadPics({ handleSelecFile }) {
```
Unused prop

DavidDavydovTech commented on src/components/qa/subcomponents/UploadPics.jsx:
```diff
@@ -0,0 +1,63 @@
+export default function UploadPics({ handleSelecFile }) {
+  const [selectedFile, setSelectedFile] = useState([]);
+  const fileSelectedHandler = (e) => {
+    let file = e.target.files[0];
+    let reader = new FileReader();
+    var url = reader.readAsDataURL(file);
```
Never reassigned, use const. `url` is unused.

DavidDavydovTech commented on src/components/qa/subcomponents/UploadPics.jsx:
```diff
@@ -0,0 +1,63 @@
+    let file = e.target.files[0];
+    let reader = new FileReader();
+    var url = reader.readAsDataURL(file);
+    reader.onloadend = (e) => {
```
Unused parameter

DavidDavydovTech commented on src/components/qa/subcomponents/Searchbar.jsx:
```diff
@@ -0,0 +1,32 @@
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+
+export default function Searchbar({handleSearchChange, value}) {
```
Missing Props Validation

DavidDavydovTech commented on src/components/qa/subcomponents/QuestionModal.jsx:
```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from 'react';
```
`useEffect` is unused.

DavidDavydovTech commented on src/components/qa/subcomponents/QuestionModal.jsx:
```diff
@@ -0,0 +1,117 @@
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import axios from 'axios'
+import { FormatListNumberedRtlOutlined } from '@material-ui/icons';
```
`FormatListNumberedRtlOutlined` is an unused import

DavidDavydovTech commented on src/components/qa/subcomponents/QuestionModal.jsx:
```diff
@@ -0,0 +1,117 @@
+import axios from 'axios'
+import { FormatListNumberedRtlOutlined } from '@material-ui/icons';
+
+export default function QuestionModal({questionsData, dataBack}) {
+  const [prod,setProd] = useState(dataBack)
+  const [formData, setFormData] = useState({
+    question: '',
+    nickname: '',
+    email: ''
+  })
+  const [productTitle,setProductTitle] = useState('')
+  const handleInputChange = (e) => {
+    setFormData({...formData,[e.target.name]: e.target.value})
+  }
+  const request = () => {
+    return axios.get(`http://18.224.37.110/products/${dataBack}`)
+      .then((data) => {
+        setProductTitle(data.data.name)
+      })
+  }
+  const handleSubmitForm = () => {
+    setFormData({
+      question: '',
+      nickname: '',
+      email: ''
+    })
+  }
+  const [open, setOpen] = useState(false);
+  const handleClickOpen = () => {
+    setOpen(true);
+    request()
+  };
```
Lots of missing semicolins and trailing commas

DavidDavydovTech commented on src/components/qa/subcomponents/QuestionModal.jsx:
```diff
@@ -0,0 +1,117 @@
+        <DialogTitle>Ask Your Question</DialogTitle>
+        <DialogContent>
+          <DialogContentText>
+            {productTitle ? productTitle : null}
```
if `productTitle` is undefined then there will be nothing to render. This code could be acomplished by using `{productTitle}` instead.

DavidDavydovTech commented on src/components/qa/subcomponents/QuestionModal.jsx:
```diff
@@ -0,0 +1,117 @@
+            value={formData.question}
+            margin="dense"
+            multiline
+            name='question'
```
string values in JSX tag use double-quotes (`"`) to keep in line with HTML standards.

DavidDavydovTech commented on src/components/qa/subcomponents/QuestionModal.jsx:
```diff
@@ -0,0 +1,117 @@
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={() => setOpen(false)}> Cancel </Button>
+          <Button onClick={() =>{
+            handleSubmitForm()
+            setOpen(false)
+            }}> Submit </Button>
```
Bad formatting. Submit and the closing tag should all be on their own line.

DavidDavydovTech commented on src/components/qa/subcomponents/QAContainer.jsx:
```diff
@@ -0,0 +1,107 @@
+import React, { useState, useEffect } from 'react';
+import ContainerQuestionEntryItem from './ContainerQuestionEntryItem';
+import { Typography } from '@material-ui/core';
```
Module imports should always happen before component imports. Switch these imports around.

DavidDavydovTech commented on src/components/qa/subcomponents/QAContainer.jsx:
```diff
@@ -0,0 +1,107 @@
+import React, { useState, useEffect } from 'react';
+import ContainerQuestionEntryItem from './ContainerQuestionEntryItem';
+import { Typography } from '@material-ui/core';
+import MoreAnsweredQuestions from './MoreAnsweredQuestions.jsx'
+import Searchbar from './Searchbar.jsx';
+import QuestionModal from './QuestionModal.jsx'
```
Remove tailing `.jsx` from these imports

DavidDavydovTech commented on src/components/qa/subcomponents/QAContainer.jsx:
```diff
@@ -0,0 +1,107 @@
+  useEffect(() => {
+    setNewData(questionsData)
+  },[addMore])  
+  // console.log("this is questions Data", questionsData)
+    // const [loadMoreClicked, setLoadMoreClicked] = useState(false)
+    // console.log("this is questions data", questionsData)
+    // const [questionData,setQuestionData] = useState(questionsData)
+    // const [searchTerm, setSearchTerm] = useState('')
+    // const [searchResults, setSearchResults] = useState(null)
+    // useEffect(() => {
+            
+    //         let results = (questionsData.slice().filter((question) => {
+    //             console.log("this is question",question.question_body)
+    //             question.question_body.toLowerCase().includes(searchTerm)
+    //         }))
+        
+    //     setSearchResults(results)
+    // },[searchTerm])
+
+    // const handleSearchChange = (e) => {
+    //     let value = e.target.value
+    //     setSearchTerm(value)
+    //   }
```
Delete these comments.

DavidDavydovTech commented on src/components/qa/subcomponents/QAContainer.jsx:
```diff
@@ -0,0 +1,107 @@
+    //   }
+   
+      
+    const renderContainerEntryItemFourQuestions = (
+        <div>
+            
+          {newData.length > 0
+            ? newData.slice().sort((a, b) => {
+                return b.question_helpfulness - a.question_helpfulness;
+              }).map((question) => {
+                  const questionsArray = Object.entries(question.answers);
+                  const answerList = questionsArray.map((item) => {
+                    // console.log('this should be each question', item);
+                    return item[1];
+                  });
+                //   console.log('this is the array of questions', answerList);
+                  return (
+                    <ContainerQuestionEntryItem
+                      questionItem={question}
+                      answers={answerList}
+                      loadMoreAnswersClicked={loadMoreAnswersClicked}
+                      dataBack={dataBack}
+                    />
+                  );
+                })
+            : null}
+        </div>
+      );
```
Incorrect indentation

DavidDavydovTech commented on src/components/qa/subcomponents/QAContainer.jsx:
```diff
@@ -0,0 +1,107 @@
+    <div>
+      {newData.length > 0
+        ? newData.slice()
+            .sort((a, b) => {
+              return b.question_helpfulness - a.question_helpfulness;
+            })
+            .map((question) => {
+              const questionsArray = Object.entries(question.answers);
+              const answerList = questionsArray.map((item) => {
+                // console.log('this should be each question', item);
+                return item[1];
+              });
+            //   console.log('this is the array of questions', answerList);
+              return (
+                <ContainerQuestionEntryItem
+                  questionItem={question}
+                  answers={answerList}
+                  loadMoreAnswersClicked={loadMoreAnswersClicked}
+                  dataBack={dataBack}
+                />
+              );
+            })
```
Incorrect indentation

DavidDavydovTech commented on src/components/qa/subcomponents/QAContainer.jsx:
```diff
@@ -0,0 +1,107 @@
+        : null}
+    </div>
+  );
+const renderQuestions = loadMoreState ? renderContainerEntryItemForAllQuestions : renderContainerEntryItemFourQuestions
```
Unused variable

DavidDavydovTech commented on src/components/qa/subcomponents/QAContainer.jsx:
```diff
@@ -0,0 +1,107 @@
+const renderQuestions = loadMoreState ? renderContainerEntryItemForAllQuestions : renderContainerEntryItemFourQuestions
+const loadMore = loadMoreAnswersClicked ? <span>Collapse answers</span> : <span>See more answers</span>
+return (
+    <>
+    <Searchbar handleSearchChange={handleSearchChange} value={searchTerm}/>
+    <br/>
+    <br/>
+    {renderContainerEntryItemFourQuestions}
+      {/* {renderContainerEntryItemForAllQuestions} */}
+      {/* {renderContainerEntryItemFourQuestions} */}
+      {/* {loadMoreClicked ? {renderContainerEntryItemForAllQuestions} : {renderContainerEntryItemFourQuestions}} */}
+      <Typography>
+        <b style={{ cursor: 'pointer' }} onClick={() => setLoadMoreAnswersClicked(!loadMoreAnswersClicked)}>{loadMore}</b>
+      </Typography>
+      <MoreAnsweredQuestions handleClick={addMore}/>
+      <QuestionModal  dataBack={dataBack}/>
+    </>
```
Bad indentation

DavidDavydovTech commented on src/components/qa/subcomponents/MoreAnsweredQuestions.jsx:
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import { Grid, Button } from '@material-ui/core';
+
+export default function MoreAnsweredQuestions({showMoreQuestions, addMore, handleClick}) {
```
Missing props validation, `showMoreQuestion` and `addMoreQuestions` are unused.

DavidDavydovTech commented on src/components/qa/subcomponents/MoreAnsweredQuestions.jsx:
```diff
@@ -0,0 +1,15 @@
+export default function MoreAnsweredQuestions({showMoreQuestions, addMore, handleClick}) {
+   
+    return (
+        <div>
+            <Grid xs='auto'>
+            <Button variant="outlined" color="primary" onClick={handleClick}>
+        MORE ANSWERED QUESTIONS 
+      </Button>
+      </Grid>
+        </div>
```
Bad indentation

DavidDavydovTech commented on src/components/qa/subcomponents/ContainerQuestionEntryItem.jsx:
```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react';
+import AnswerListEntryItem from './AnswerListEntryItem';
+import AnswerModal from './AnswerModal.jsx'
+import { Grid, Typography } from '@material-ui/core';
```
All module imports should happen before component imports

DavidDavydovTech commented on src/components/qa/subcomponents/ContainerQuestionEntryItem.jsx:
```diff
@@ -0,0 +1,86 @@
+
+
+export default function ContainerQuestionEntryItem({ questionItem, answers,loadMoreAnswersClicked, dataBack }) {
+  const [twoAnswersArray, setTwoAnswersArray] = useState(answers.slice(0, 2));
+  const [allAnswers, setAllAnswers] = useState(answers);
```
Unused states / setters

DavidDavydovTech commented on src/components/qa/subcomponents/ContainerQuestionEntryItem.jsx:
```diff
@@ -0,0 +1,86 @@
+        <Grid xs={3} item>
+          <Typography color="textSecondary" variant="caption">
+          <span
+            className="helpful_status"
+           
+          >
+            {' '}
+            Helpful? <u style={{cursor: 'pointer'}} onClick={handleVoteClicked}>Yes</u> (
+            {voteCount}){' '}    |    <u style={{cursor: 'pointer'}}><AnswerModal questionItem={questionItem} dataBack={dataBack}/></u>
+          </span>
```
Code is hard to parse. Clean it up to make ti more readable.

DavidDavydovTech commented on src/components/qa/subcomponents/ContainerQuestionEntryItem.jsx:
```diff
@@ -0,0 +1,86 @@
+          </Typography>
+        </Grid>
+      </Grid>
+      {/* <div className="question_title">
+        <span style={{ fontWeight: 'bold' }}>
+          Q: {questionItem.question_body}
+        </span>
+        <span
+          className="helpful_status"
+          style={{ display: 'inline-block', marginLeft: '300px' }}
+        >
+          {' '}
+          Helpful? <span style={{ textDecoration: 'underline' }}>Yes</span> (
+          {questionItem.question_helpfulness}){' '}
+        </span>
+      </div>
+      <div className="answers_component">{renderAllAnswers}</div>
+    </div> */}
```
Remove this comment.

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerModal.jsx:
```diff
@@ -0,0 +1,125 @@
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import UploadPics from './UploadPics.jsx';
+import {Typography} from '@material-ui/core';
+import UploadMediaCard from './UploadMediaCard';
+import axios from 'axios'
```
Modules should always be imported before components

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerModal.jsx:
```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
```
this should all be in a single import from `@material-ui/core` using destructoring

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerModal.jsx:
```diff
@@ -0,0 +1,125 @@
+import UploadMediaCard from './UploadMediaCard';
+import axios from 'axios'
+
+export default function AnswerModal({dataBack, questionItem}) {
+  const [productTitle,setProductTitle] = useState('');
+  const [formData, setFormData] = useState({
+    answer: '',
+    nickname: '',
+    email: ''
+  })
+  const [open, setOpen] = useState(false);
+  const request = () => {
+    console.log("this is db", dataBack)
+    return axios.get(`http://18.224.37.110/products/${dataBack}`)
+      .then((data) => {
+        console.log(data)
+        setProductTitle(data.data.name)
+      })
+      .catch((err) => console.log(err))
+  }
+  const handleInputChange = (e) => {
+    setFormData({...formData,[e.target.name]: e.target.value})
+  }
+  const handleClickOpen = () => {
+    setOpen(true);
+    request();
+  };
+  const handleSubmitForm = () => {
+    setFormData({
+      answer: '',
+      nickname: '',
+      email: ''
+    })
+  }
```
Minor styling errors (missing semicolons, trailing commas.

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerModal.jsx:
```diff
@@ -0,0 +1,125 @@
+    })
+  }
+  return (
+    <>
+    <Typography color="textSecondary" variant="caption" >
```
Bad indentation

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerModal.jsx:
```diff
@@ -0,0 +1,125 @@
+  return (
+    <>
+    <Typography color="textSecondary" variant="caption" >
+      <span onClick={handleClickOpen}>
+        Add Answer
+      </span>
+      </Typography>
+      <Dialog open={open}>
+      <input
+            accept="image/*"
+            style={{ display: 'none' }}
+            id="raised-button-file"
+            multiple
+            type="file"
+          />
+        <DialogTitle>Submit Your Answer</DialogTitle>
+        <DialogContent>
```
General styling errors

DavidDavydovTech replied:
P.S. check the linter.

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerModal.jsx:
```diff
@@ -0,0 +1,125 @@
+          />
+        <DialogTitle>Submit Your Answer</DialogTitle>
+        <DialogContent>
+        <DialogContentText>{productTitle ? <b>{productTitle}</b> : null}: {questionItem ? questionItem.question_body : null}</DialogContentText>
```
Hard to parse. Consider making this multi-line.

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerModal.jsx:
```diff
@@ -0,0 +1,125 @@
+          <Button onClick={() =>{
+            handleSubmitForm()
+            setOpen(false)
+            }}> Submit </Button>
```
Submit and the closing tag should be on their own lines.

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerListEntryItem.jsx:
```diff
@@ -0,0 +1,106 @@
+import React, {useState} from 'react';
+import {Card, CardMedia, Typography, Paper} from '@material-ui/core';
```
Card is an unused import

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerListEntryItem.jsx:
```diff
@@ -0,0 +1,106 @@
+  const [voteClicked, setVoteClicked] = useState(false)
+  const [voteCount, setVoteCount] = useState(answer.helpfulness)
+  const handleVoteClicked = () => {
+    if(voteClicked){
+      return <>Already Voted</>
+    } else {
+      setVoteCount(() => voteCount + 1)
+    setVoteClicked(true)
+    }
+    
```
General bad styling (check linter).
Else is unneeded because of the return statement inside the `if` block.

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerListEntryItem.jsx:
```diff
@@ -0,0 +1,106 @@
+    } ${converted.getDay()}, ${converted.getFullYear()}`;
+  };
+  const renderPhotos = answer.photos.map((photo) => {
+    const { answerer_name, date, helpfullness } = answer;
```
Unused variables

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerListEntryItem.jsx:
```diff
@@ -0,0 +1,106 @@
+      </>
+    );
+  });
+  const renderPhotoMessage =
+    answer.photos.length > 0 ? (<>
+      <div className="render_photo_message">
+        <span>{renderPhotos}</span>
+      </div>
+        </>) : null;
```
A little messy to visually parse, consider cleaning up

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerListEntryItem.jsx:
```diff
@@ -0,0 +1,106 @@
+      //   <h5>I am a photo</h5>
+      <>
+      <Paper 
+        elevation={0}
+
+      >
```
Unneeded whitespace

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerListEntryItem.jsx:
```diff
@@ -0,0 +1,106 @@
+        color="textSecondary"
+        variant="caption"
+      >
+        {`by ${answer.answerer_name}, ${formatDate(answer.date)}  |  Helpful? `}
+        {/* by {answer.answerer_name}, {formatDate(answer.date)} | Helpful?{' '} */}
+        <u style={{cursor: 'pointer'}} onClick={handleVoteClicked}>Yes</u> 
+        {`(${voteCount})  |  ` }
+        <span onClick={toggleReport} style={{cursor: 'pointer'}}>{report}</span>
```
Visually hard to parse. Open up the `span` on line `98`, delete the comment on line `95`.

DavidDavydovTech commented on src/components/qa/subcomponents/AnswerListEntryItem.jsx:
```diff
@@ -0,0 +1,106 @@
+  );
+}
+
+// <Typography className={classes.title} color="textSecondary" gutterBottom>
+//           Word of the Day
+//         </Typography>
```
Delete this comment

Commit:
Quick Fixes

This is a band-aid commit that merges with master, makes minimal fixes,
and allows us to continue and finish within the deadline.
